Add 'separate' list filter

drybjed opened the pull request:
'separate' filter allows you to convert a YAML list into a string with
each list member separated by a specified string.

Example usage:

```

---
- hosts: all
  vars:
    gpg_recipients: [ pubkey1, pubkey2, pubkey3 ]

  tasks:
    - name: Encrypt file to multiple recipients
      command: gpg --encrypt -r {{ gpg_recipients|separate(' -r ') }} -a /path/to/file creates=/path/to/file.asc
```

drybjed committed:
Add 'separate' list filter

'separate' filter allows you to convert a YAML list into a string with
each list member separated by a specified string.

mpdehaan commented:
Seems like split and join already provide these things.

http://jinja.pocoo.org/docs/templates/#builtin-filters

foo.split(",") | join(":")

etc

?

drybjed commented:
@mpdehaan Yes, but it works inside templates. I'm not sure how I should use it in a command: task? I tried this:

```
---

- hosts: all

  vars:
    gpg_keys: [ key1, key2, key3 ]
    gpg_recipients:
      - 'key4'
      - 'key5'
      - 'key6'

  tasks:
    - debug: msg="gpg --encrypt -r {{ gpg_keys|separate(' -r ') }} --armor file"
    - debug: msg="gpg --encrypt -r {{ gpg_recipients|separate(' -r ') }} --armor file"
    - debug: msg="gpg --encrypt -r {{ gpg_keys|split(',') | join(' -r ') }} --armor file"
    - debug: msg="gpg --encrypt -r {{ gpg_recipients|split(',') | join(' -r ') }} --armor file"
    - debug: msg="gpg --encrypt -r {{ gpg_keys.split(',') | join(' -r ') }} --armor file"
```

And the results:

```
PLAY [all] ******************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [sks01.aiua]

TASK: [debug msg="gpg --encrypt -r key1 -r key2 -r key3 --armor file"] ******** 
ok: [sks01.aiua] => {
    "msg": "gpg --encrypt -r key1 -r key2 -r key3 --armor file"
}

TASK: [debug msg="gpg --encrypt -r key4 -r key5 -r key6 --armor file"] ******** 
ok: [sks01.aiua] => {
    "msg": "gpg --encrypt -r key4 -r key5 -r key6 --armor file"
}

TASK: [debug msg="gpg --encrypt -r {{gpg_keys|split(',') | join(' -r ')}} --armor file"] *** 
ok: [sks01.aiua] => {
    "msg": "gpg --encrypt -r {{gpg_keys|split(',') | join(' -r ')}} --armor file"
}

TASK: [debug msg="gpg --encrypt -r {{gpg_recipients|split(',') | join(' -r ')}} --armor file"] *** 
ok: [sks01.aiua] => {
    "msg": "gpg --encrypt -r {{gpg_recipients|split(',') | join(' -r ')}} --armor file"
}

TASK: [debug msg="gpg --encrypt -r {{gpg_keys.split(',') | join(' -r ')}} --armor file"] *** 
fatal: [sks01.aiua] => One or more undefined variables: 'list object' has no attribute 'split'

FATAL: all hosts have already failed -- aborting
```

How do you use your example in a task?

drybjed commented:
@lied found the solution on IRC:

```
{{ gpg_keys | join(' -r ') }}
```
